VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleListEntryController.cs

```diff
@@ -217,6 +209,100 @@ public IHttpActionResult Move(webModel.MoveInfo moveInfo)
+            var productIds = new List<string>();
+            var categoryIds = new List<string>();
+
+            if (idsToDelete.IsNullOrEmpty())
+            {
+                idsToDelete = GetIdsToDelete(searchCriteria);
+            }
+
+            if (!idsToDelete.IsNullOrEmpty())
+            {
+                idsToDelete.ProcessWithPaging(DeleteBatchSize, (ids, currentItem, totalCount) =>
+                {
+                    var commonIds = ids.ToArray();
+                    var searchProductResult = _itemService.GetByIds(commonIds, coreModel.ItemResponseGroup.None);
+                    CheckCurrentUserHasPermissionForObjects(CatalogPredefinedPermissions.Delete, searchProductResult);
+                    productIds.AddRange(searchProductResult.Select(x => x.Id));
+
+                    var searchCategoryResult = _categoryService.GetByIds(commonIds.ToArray(), coreModel.CategoryResponseGroup.None);
+                    CheckCurrentUserHasPermissionForObjects(CatalogPredefinedPermissions.Delete, searchCategoryResult);
+                    categoryIds.AddRange(searchCategoryResult.Select(x => x.Id));
+                });
+
+                productIds.ProcessWithPaging(DeleteBatchSize, (ids, currentItem, totalCount) =>
+                {
+                    _itemService.Delete(ids.ToArray());
+                });
+
+                categoryIds.ProcessWithPaging(DeleteBatchSize, (ids, currentItem, totalCount) =>
+                {
+                    _categoryService.Delete(ids.ToArray());
+                });
+            }
+
+            return StatusCode(HttpStatusCode.NoContent);
+        }
+
+
+        private coreModel.SearchCriteria PrepareSearchCriteria(webModel.SearchCriteria criteria)
+        {
+            var coreModelCriteria = criteria.ToCoreModel();
+            ApplyRestrictionsForCurrentUser(coreModelCriteria);
+
+            coreModelCriteria.WithHidden = true;
+
+            // need search in children categories if user specify keyword
+            if (!string.IsNullOrEmpty(coreModelCriteria.Keyword))
+            {
+                coreModelCriteria.SearchInChildren = true;
+                coreModelCriteria.SearchInVariations = true;
+            }
+
+            return coreModelCriteria;
+        }
+
+        private List<string> GetIdsToDelete(webModel.SearchCriteria searchCriteria)
+        {
+            // Any pagination for deleting should be managed at back-end. 
+            searchCriteria.Take = DeleteBatchSize;
+            searchCriteria.Skip = 0;
+
+            var itemIds = new List<string>();
+            bool hasItems;
+
+            do
+            {
+                var searchResult = _listEntrySearchService.Search(PrepareSearchCriteria(searchCriteria));
+                var listEntriesIds = searchResult.ListEntries.Where(x => string.Equals(x.Type, KnownDocumentTypes.Product, StringComparison.InvariantCultureIgnoreCase) ||
```

Comment:
please use `EqualsInvariant` extension, it is much shorter

Reply:
used